### PR TITLE
[Merged by Bors] - Add support for Google sheets connector

### PR DESCRIPTION
**Pull request by sbernauer**
Closes https://github.com/stackabletech/trino-operator/issues/339

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/catalog/config.rs [hunk 1]
```diff
@@ -104,6 +104,11 @@ impl CatalogConfig {
                     .to_catalog_config(&catalog_name, catalog_namespace, client)
                     .await
             }
+            TrinoCatalogConnector::GoogleSheet(connector) => {
```
Comment (adwk67) [inline rust/operator-binary/src/catalog/config.rs:107]: Just curious: would there be any advantage in having a trait for `to_catalog_config`, so the `match` statement wouldn't have to changed every time a new enum variant is added? (I like the enum pattern but am not deep enough into Rust to know the pros/cons).

**Review by adwk67 — APPROVED** [review]
LGTM % missing-trino-link

File: docs/modules/ROOT/pages/concepts.adoc [hunk 2]
```diff
@@ -31,4 +31,4 @@ The following diagram illustrates this. Two Trino catalogs - each an instance of
 include::partial$diagrams/TrinoCatalogs.excalidraw[]
 ----
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply docs/modules/ROOT/pages/concepts.adoc:34]: Good catch!

File: rust/operator-binary/src/catalog/config.rs (hunk 1 above)
Comment (adwk67) [reply rust/operator-binary/src/catalog/config.rs:107]: That's fine by me

**Review by adwk67 — APPROVED** [review]

**Pull request closed without merging**